tmt/utils/__init__.py

```diff
@@ -4358,6 +4358,30 @@ def increment(
 
                 raise GeneralError(message) from error
 
+        # Handle GitHub-specific responses
+        # https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
+        response = cast(requests.Response, kwargs.get('response'))
+        if response is not None and 'X-GitHub-Request-Id' in response.headers:
```

Comment:
@martinhoyer by negating this condition you could save an indendation below I believe

```
        if response is None or 'X-GitHub-Request-Id' not in response.headers:
                  return super().increment(*args, **kwargs)
                  
        headers = response.headers
        ....
```

Reply:
smart